src/lock/file.rs

```diff
@@ -58,6 +58,10 @@ impl LockedConfig {
         P: AsRef<Path>,
     {
         let path = path.as_ref();
+        if let Some(parent) = path.parent() {
+            std::fs::create_dir_all(parent)
+                .with_context(s!("failed to write locked config to `{}`, because the parent directory couldn't be created", path.display()))?;
+        }
```

Comment:
this is new, because without it, sheldon complained to me that it couldn't lock the plugins. Not sure if this is the best place, but it's at least a good enough place.

Reply:
This seems fine. The calling function already gives some context so I think you just make the context here

```rust
s!("failed to create parent directory for `{}`, path.display())
```

Reply:
Okay, adapted the context.